File: c/p2.c
//https://projecteuler.net/problem=2
//Even Fibonacci numbers

#include <stdio.h>

int main(void) {
    int n = 4000000;
    int sum = 0;
    int curr = 2;
    int prev = 1;
    while (curr <= n) {
        if (curr % 2 == 0) {
            sum += curr;
        }
        int s = curr + prev;
        prev = curr;
        curr = s;
    }
    printf("%d\n", sum);
}

File: python/p2.py
#https://projecteuler.net/problem=2
#Even Fibonacci Numbers

a,b = 1,2
s = 2
k = -1
while b<4000000:
    tmp = a+b
    a=b
    b=tmp
    if b%2 == 0:
        s+=b
print(s)